src/docfx/build/toc/parser/TableOfContentsParser.cs

```diff
@@ -15,13 +15,14 @@ internal static class TableOfContentsParser
         private static readonly string[] s_tocFileNames = new[] { "TOC.md", "TOC.json", "TOC.yml" };
         private static readonly string[] s_experimentalTocFileNames = new[] { "TOC.experimental.md", "TOC.experimental.json", "TOC.experimental.yml" };
 
-        public delegate (string resolvedTopicHref, List<string> monikers) ResolveHref(Document relativeTo, string href, Document resultRelativeTo);
+        public delegate (Error error, string href, Document file) ResolveHref(Document relativeTo, string href, Document resultRelativeTo);
 
-        public delegate (string content, Document file) ResolveContent(Document relativeTo, string href, bool isInclusion);
+        public delegate (Error error, string content, Document file) ResolveContent(Document relativeTo, string href);
```

Comment:
> public delegate (Error error, string content, Document file) ResolveContent(Document relativeTo, string href); [](start = 7, length = 111)

if you want to make it consistent with BuildPage, do you want move this delegate out? You are trying to clean the interface? but actually finally we all call TryResolveContent with some wrapper?

Reply:
then if we always need the wrapper, we can be different right?

---
In reply to: [238950245](https://github.com/dotnet/docfx/pull/3722#discussion_r238950245) [](ancestors = 238950245)